web/app.js

```diff
@@ -867,6 +868,15 @@ const PDFViewerApplication = {
       typeof PDFJSDev === "undefined" ||
       PDFJSDev.test("MOZCENTRAL || GENERIC")
     ) {
+      if (featureId && !this._unsupportedFeatureErrors.includes(featureId)) {
```

Comment:
null check for `featureId` because of the changes made in https://github.com/mozilla/pdf.js/pull/11953. That would trigger fallback() with a null featureId

Reply:
Unless I'm mistaken, wouldn't it be a lot simpler/safer to do this kind of validation/tracking of seen `featureId`s in the new case that you're adding to [`ChromeActions.reportTelemetry`](https://searchfox.org/mozilla-central/rev/598e50d2c3cd81cd616654f16af811adceb08f9f/browser/extensions/pdfjs/content/PdfStreamConverter.jsm#424) and simply report unconditionally here instead (note how e.g. `documentStats` is implemented)?

Since that's the central spot where all telemetry results pass through, you'd be thus be guaranteed to only report a specific `featureId` value *once*.
By tracking seen `featureId`s in manually in `web/app.js`, as is done in this patch, it seems much easier for any future changes to accidentally e.g. double-report telemetry data if one forgets to add a `this._unsupportedFeatureErrors.push(featureId);` line.